src/link.zig

```diff
@@ -435,6 +435,24 @@ pub const File = struct {
         }
     }
 
+    /// Called from within CodeGen to retrieve the symbol index of a global symbol.
+    /// If no symbol exists yet with this name, a new one will be created instead.
+    pub fn getGlobalSymbol(base: *File, name: []const u8) UpdateDeclError!u32 {
+        log.debug("getGlobalSymbol '{s}'", .{name});
+        switch (base.tag) {
+            // zig fmt: off
+            .coff  => unreachable,
+            .elf   => unreachable,
+            .macho => unreachable,
```

Comment:
This is a big ask so feel free to refuse it - MachO already implements this function but under a different name `addExternFn`. Would you be OK with renaming it to `getGlobalSymbol` in MachO and plugging it here too?

Reply:
Thank you kindly!